scripts/model_upload.py

```diff
@@ -31,14 +33,79 @@ def upload_part(url, params, headers, data, parts, verify):
     parts.append({'ETag': part_etag, 'PartNumber': params['partNumber']})
 
 
+def upload(url, params, headers, part_size, verify=False):
+    file_path = params['file_path']
+    if os.path.isfile(file_path):
+        if tarfile.is_tarfile(file_path):
+            untar_and_upload(url, params, headers, part_size, verify)
+        else:
+            upload_model(url, params, headers, part_size, verify)
+    elif os.path.isdir(file_path):
+        upload_dir(url, params, headers, part_size, verify)
+    else:
+        raise Exception("Unrecognized type of upload")
+
+
+def untar_and_upload(url, params, headers, part_size, verify=False):
+    tar = tarfile.open(params['file_path'])
+    tar.extractall(path='/tmp')
+    tar.close()
+    tmp_path = '{}{}'.format('/tmp', tar.members[0].path.strip('.'))
+    params['file_path'] = tmp_path
+    upload_dir(url, params, headers, part_size, verify)
+    shutil.rmtree(tmp_path, ignore_errors=True)
+
+
+def upload_dir(url, params, headers, part_size, verify=False):
+    root_dir_name = params['file_path'].split('/')[-1]
```

Comment:
what if, there will be another `dir` in this `dir`?
Maybe this method should be recursive?

Reply:
I assumed that one will pass a dir with typical structure of dir as in TFS. Dealing with dirs and thinking of what user would like to upload and how it will look is not trivial, so I made such assumptions